### PR TITLE
feature/add raw efile filters

### LINKED ISSUE #3813: Add filters to /schedules/schedule_e/efile
(/schedules/schedule_e/efile/)

So that we can implement fec-cms/#2899 we need to add filters to the API

Filter options should include
- [x] Spender name
- [x] Candidate name or ID
- [x] Support/Oppose
- [x] Candidate office state
- [x] Candidate office district
- [x] Candidate party
- [x] Candidate office
- [x] Filing date (F24 or F5 filing date)
- [x] Expenditure date
- [x] (maybe) Dissemination date
- [x] Most recent (match filings filter)

**Completion Criteria**
- [x] Figure out how difficult these filters are to add - are some easier than others?
- [x] Add the filters
- [x] Add tests for each one of the filter

### PR DESCRIPTION
**Pull request by pkfec**
## Summary (required)

- Resolves #3813 

_Include a summary of proposed changes._

## How to test the changes locally

- checkout the branch `feature/add-efile-filters`
- Update the itemized.py SE model class to point to `real_efile_se_f57_vw_tmp_pk` tmp view here: https://github.com/fecgov/openFEC/blob/feature/add-raw-efile-filters/webservices/common/models/itemized.py#L659
- export SQLA_CONN to point to DEV DB
- start server ./manage.py runserver
- on Swagger API specs, go to Independent Expenditure section, 
`/schedules/schedule_e/efile/` endpoint.
- test the filters.

## Impacted areas of the application
- Swagger api spec
-  Independent Expenditures raw efilings

### CODE REVIEW

File: webservices/common/models/itemized.py [hunk 1]
```diff
@@ -652,7 +652,8 @@ class ScheduleE(PdfMixin, BaseItemized):
 
 
 class ScheduleEEfile(BaseRawItemized):
-    __tablename__ = 'real_efile_se_f57_vw'
+    # __tablename__ = 'real_efile_se_f57_vw'
+    __tablename__ = 'real_efile_se_f57_vw_tmp_pk'
```
Comment (hcaofec) [inline webservices/common/models/itemized.py]: I think we should use real view, not the _tmp

**Comment by hcaofec** [comment]
migration file for adding tsvector column is missing
@pkfec

File: webservices/common/models/itemized.py (hunk 1 above)
Comment (pkfec) [reply webservices/common/models/itemized.py]: Thanks @hcaofec for reviewing. This is a WIP PR and not ready for review yet. Paul is testing the  SE efile new tsvector filter on his local off my feature branch. I will update and upload the latest view after i get a sign off from Paul.

**Comment by pkfec** [comment]
@hcaofec @fec-jli Please review.

File: webservices/args.py [hunk 2]
```diff
@@ -797,10 +797,9 @@ def make_seek_args(field=fields.Int, description=None):
 }
 
 schedule_e_efile = {
+    'candidate_search': fields.List(fields.Str, description=docs.CANDIDATE_ID),
     'committee_id': fields.List(IStr, description=docs.COMMITTEE_ID),
-    'candidate_id': fields.List(IStr, description=docs.CANDIDATE_ID),
     'payee_name': fields.List(fields.Str, description='Name of the entity that received the payment'),
```
Comment (fec-jli) [inline webservices/args.py]: Can you create doc variables to replace `description` ?

File: webservices/args.py [hunk 3]
```diff
@@ -797,10 +797,9 @@ def make_seek_args(field=fields.Int, description=None):
...
-    'candidate_name': fields.List(fields.Str, description=docs.CANDIDATE_NAME),
     'image_number': fields.List(
         fields.Str,
         description='The image number of the page where the schedule item is reported',
```
Comment (fec-jli) [inline webservices/args.py]: Can you create doc variables to replace `description` ?

File: webservices/args.py [hunk 4]
```diff
@@ -809,10 +808,20 @@ def make_seek_args(field=fields.Int, description=None):
         IStr(validate=validate.OneOf(['S', 'O'])),
         description='Support or opposition',
```
Comment (fec-jli) [inline webservices/args.py]: Can you create doc variables to replace `description` ?

**Comment by fec-jli** [comment]
great work!
some suggestions:
1)in args.py --schedule_e_efile section, can you create some doc variables to replace `description`?
2)in swagger page, for filter: **candidate_search**.  it seems we should make description more clear for search "candidate name or candidate_id" 
3)for function **def name_generator(*args):**,  it is better move to utils file, (may create a separated ticket to work on this)

**Comment by hcaofec** [comment]
Test this url and the results include Hillary data
http://127.0.0.1:5000/v1/schedules/schedule_e/efile/?page=1&candidate_office=P&sort_null_only=false&sort=-expenditure_date&per_page=20&sort_hide_null=false&support_oppose_indicator=S&candidate_name=TRUMP&sort_nulls_last=false

**Pull request merged**